app/models/whitehall_migration/timeline_entry.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+class WhitehallMigration::TimelineEntry < ApplicationRecord
+  enum entry_type: {
+    archived: "archived",
+    document_updated: "document_updated",
+    fact_check_request: "fact_check_request",
+    fact_check_response: "fact_check_response",
+    first_created: "first_created",
+    imported_from_whitehall: "imported_from_whitehall",
+    internal_note: "internal_note",
+    new_edition: "new_edition",
+    published: "published",
+    rejected: "rejected",
+    removed: "removed",
+    scheduled: "scheduled",
+    submitted: "submitted",
+    withdrawn: "withdrawn",
+  }
```

Comment:
We should flag this class as read_only after creation since we wouldn't expect anyone to have reason to edit it under the normal application flow: 

https://github.com/alphagov/content-publisher/blob/master/app/models/revision.rb#L67-L69

Reply:
Done